### ISSUE TITLE
Expose `wait_readable` and `wait_writable` in SSLSocket

### ISSUE THREAD

**Issue by ioquatix** [ruby/openssl#408]
It seems like we don't expose this.

It was a problem for this PR: https://github.com/redis/redis-rb/pull/960

User should be able to expect to write:

```
io.wait_readable(timeout)
io.wait_writable(timeout)
```

But it doesn't seem to be working for `SSLSocket`.

It should be relatively trivial to implement, something like this:

```
        def wait_readable(timeout = nil)
          to_io.wait_readable(timeout)
        end

        def wait_writable(timeout = nil)
          to_io.wait_writable(timeout)
        end
```

Any thoughts?

**Comment by rhenium** [comment]
Agreed, that would be a nice addition. Other methods from io/wait (#nread - can simply return 0, #ready?, #wait, #wait_priority) can also be implemented in a similar way.

**Comment by djberg96** [comment]
Has this still not happened yet?

**Comment by rhenium** [comment]
https://github.com/ruby/openssl/pull/708 implemented them (in 2024).

**Comment by ioquatix** [comment]
Thanks!
